test/daml/Test/Lifecycle.daml

```diff
@@ -290,11 +290,11 @@ testAmericanPut = script do
 
   -- You must first exercise the 'outer' `anytime`, and then ...
   remaining <- Lifecycle.exercise observe25 (bearer, remaining.claim) remaining t
-  remaining === until (TimeGte $ afterTomorrow) (payoff `or` zero)
+  remaining === until (TimeGte afterTomorrow) (payoff `or` zero)
 
   -- ... the inner `or`.
   remaining <- Lifecycle.exercise observe25 (bearer, payoff) remaining t
-  remaining === until (TimeGte $ afterTomorrow) payoff
+  remaining === until (TimeGte afterTomorrow) payoff
```

Comment:
This is currently failing. The question here is whether we should be reaching the inner `Anytime`. Also have a look at the pricing paper - there's an interesting observation about the fact that `Until . Anytime` is not equal to `Anytime . Until`.

Reply:
So I see the issue here now - When we are at maturity date T, and the American option is of the form `Anytime _ (Until (TimeGte T) payoff)`, we can't reach the payoff. This is likely why I didn't have `Until` respect the `acquire'` function.

I think what this boils down to is that we only have a `TimeGte` operator in this case.

So a concrete example, say `T = 1 Mar 2020` and we are at maturity date today. What this means is that `Until (T >= T) _ == Until true _`, and so the payoff will be Zero.

One solution would be to add a `TimeGt` (without the 'e') operator. I hesitate to do this, because it takes us a step up from having a preorder for time, to a total order. And I don't know how that may affect the calculus. Not to mention that it bloats the code.

Appreciate your thoughts on this @matteolimberto-da

Reply:
I guess this is also not an issue if we're using non-ordinal time, so e.g. a `Time`. For instance, if expiry has a hard cutoff (time (date 2020 Mar 1) 16 30 00), then this works correctly, I think. You must exercise before the market close, and you can expire after it correctly.

So **the problem does boil down to `Date` being a pre-order rather than an order** in the context of our problem.

Perhaps we just need to take the stance that users need to map their `Claim Date _ _ ` into a `Claim Time _ _`.

For instance, at the instrument level, you may have `Claim Date _ _`, and then you 

```haskell
mapParams
  \dt -> time dt 16 30 00
  toUTCDate
  identity
  identity
```
Now you can run `exercise` (before 16:30 UTC), and then expire (after 16:30 UTC).